### PR TITLE
Handle spot termination when a test is running.

### PR DESCRIPTION
**Pull request by dvalinrh**
Two options
1) We will create the vm without spot pricing
2) We simply terminate.

### CODE REVIEW

**Review by sousinha97 — APPROVED** [review]

File: bin/kick_off.sh [hunk 1]
```diff
@@ -119,6 +123,44 @@ do
...
+		if [ $spot_recover -eq 1 ] &&[[ ! -f "test_returned" ]]; then
+			#
+			# Check to see if we used spot, and we are to recover if the system goes away.
+			#
+			if [[ -f test_started ]]; then
+				sp_check=`grep spot_range: ansible_vars_main.yml`
+				if [[ $sp_check == *","* ]]; then
+					rm -rf tf
+					#
+					# Next attempt is with out spot pricing.
```
Comment (frival) [inline bin/kick_off.sh]: micro-nit: attempt *it*

File: bin/kick_off.sh [hunk 2]
```diff
@@ -119,6 +123,44 @@ do
...
+					#
+					grep -v "spot_range:" ansible_vars_main.yml | grep -v spot_range: > spot_repair
```
Comment (frival) [inline bin/kick_off.sh]: I'm curious why you're "grep -v"-ing twice here, the input through the pipe should be excluding spot_range: already.

File: bin/kick_off.sh [hunk 3]
```diff
@@ -119,6 +123,44 @@ do
...
+					# Not spot, test started, system died.
+					#
+					echo Error: test started, system died.
+					exit
+				fi
+			else
+				#
+				# Rarely will happen.
+				#
+				echo Error: did not start the test.
```
Comment (frival) [inline bin/kick_off.sh]: We should log something to stable storage, this echo will get lost in the massive scrollback buffer any significant run creates.

**Review by frival — CHANGES_REQUESTED** [review]
Couple issues, both "relatively" minor.

**Pull request merged**